visual.js

```diff
@@ -153,3 +106,57 @@ function drawPiece(piece) {
     pieceImages[color + piece.type] = img;
     img.setAttribute("src", fileName + ".png");
 }
+
+// Getting mouse position when canvas is clicked.
+function getMousePos(canvas, e) {
+    var rect = canvas.getBoundingClientRect();
+    return {
+        x: e.clientX - rect.left,
+        y: e.clientY - rect.top
+    };
+}
+
+let pieceToMove = null;
+
+function clickOnPiece(e) {
+    // Getting mouse position on the clicked piece.
+    let mousePos = getMousePos(canvas, e);
+    let boardFieldX = Math.ceil(mousePos.x / squareSize - 1);
+    let boardFieldY = Math.ceil(mousePos.y / squareSize - 1);
+    // Checking if piece is clicked or null
+    let currentPiece = getPieceFromXY(boardFieldX, boardFieldY);
+
+    // Returning array with allowed moves.
+    let allowedPositions = allowedMoves(boardFieldX, boardFieldY);
+
+    // Re-draw the board before adding new markers.
+    drawChessGame();
+
+    // Drawing allowed move markers on canvas.
+    allowedPositions.forEach(position => {
+        // Draw allowed position markers.
+        ctx.beginPath();
+        ctx.arc((position.x * squareSize + boardOffset) - (1 / 2 * squareSize), (position.y * squareSize + boardOffset) - (1 / 2 * squareSize), 1 / 10 * squareSize, 0, 2 * Math.PI);
+        ctx.fillStyle = "#e6ac00";
+        ctx.fill();
+    })
+
+    // Moving pieces. 
+    if (pieceToMove == null) {
+        // Select piece.
+        pieceToMove = currentPiece;
+    } else {
+        // Move piece.
+        pieceToMove.x = boardFieldX;
+        pieceToMove.y = boardFieldY;
+        pieceToMove = null;
+    }
+}
+
+function getPieceFromXY(x, y) {
+    const piece = getPieceFromPosition(x, y);
+    if (typeof piece === "object") {
+        return piece;
+    }
+    return null;
+}
```

Comment:
There is same function in gameplay.js named getPieceFromPosition which does the same thing. :)

Reply:
@austramarta why is this getPieceFromXY needed at all? Compatibility issues?

Reply:
So that I wouldn't have to always check if the piece type is an object.